### PR TITLE
💄 Add bar chart with rounded tops

### PR DESCRIPTION
**Pull request by eliascodes**
Related to #130 

Found that the code actually admitted quite a bit of re-organisation, and that a _lot_ more of it was dead code than we thought!

### Changes
* Create types for basic concepts required for drawing calculations
* Separate side-effects (drawing) from pure stuff (calculating parameters for drawing)
* Make rounding work with native legend (by inspecting hidden) as well as with custom legend (by relying on 0-values)
* `RoundedBarChart` exports a mutating function so its parent can control when the mutation occurs.

### CODE REVIEW

**Comment by astroash** [comment]
Switching between volunteers and time page causes the following error:
Screen Shot 2019-06-12 at 18 23 36

**Comment by astroash** [comment]
@eliascodes do you want to readd your no data graph view to this PR or add separately?

Screen Shot 2019-06-13 at 12 36 02

**Comment by eliascodes** [comment]
I'll do it separately

File: dashboard-app/src/components/StackedBarChart/RoundedBar/RoundedBarChart.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+export const checkShouldRound = (datasets: any[], datasetIdx: number, dataIdx: number) =>
+  !datasets
+    .slice(datasetIdx + 1)
+    .some((dataset: any) =>
+      dataset.data[dataIdx] > 0 &&
+      !(dataset._meta && dataset._meta[0] && dataset._meta[0].hidden)
+    );
+
+/*
+ * !!MUTATION!!
```
Comment (astroash) [inline dashboard-app/src/components/StackedBarChart/RoundedBar/RoundedBarChart.ts]: 🐢

**Pull request merged**